src/guidescanpy/commands/generate_kmers.py

```diff
@@ -138,15 +137,17 @@ def output_kmer(prefix, chrm_name, kmer):
 
 def output(fasta_file, args):
     print("id,sequence,pam,chromosome,position,sense")
+    kmers_count = 0
     for record in SeqIO.parse(fasta_file, "fasta"):
         if len(record) < args.min_chr_length:
             continue
-
-        for kmer in itertools.islice(
-            find_all_kmers(args.pam, args.kmer_length, record.seq, end=not args.start),
-            args.max_kmers,
+        for kmer in find_all_kmers(
+            args.pam, args.kmer_length, record.seq, end=not args.start
         ):
+            kmers_count += 1
             output_kmer(args.prefix, record.name, kmer)
+            if kmers_count >= args.max_kmers:
```

Comment:
Thanks! I think we need to take care of the case when `max_kmers` may be `None`.

Reply:
Yes, thank you! I'll fix it.